src/csharp/Microsoft.Spark/ML/Feature/Bucketizer.cs

```diff
@@ -0,0 +1,206 @@
+        /// </summary>
+        /// <param name="value">List of input columns to use as sources for buckets</param>
+        /// <returns>`Bucketizer`</returns>
+        public Bucketizer SetInputCols(List<string> value)
+        {
+            return WrapAsBucketizer(_jvmObject.Invoke("setInputCols", value));
+        }
+
+        /// <summary>
+        /// The `Bucketizer` will create a new column in the DataFrame, this is the name of the
+        /// new column.
+        /// </summary>
+        /// <param name="value">The name of the new column which contains the bucket ID</param>
+        /// <returns>`Bucketizer`</returns>
+        public Bucketizer SetOutputCol(string value)
+        {
+            return WrapAsBucketizer(_jvmObject.Invoke("setOutputCol", value));
+        }
+        
+        /// <summary>
+        /// The list of columns that the `Bucketizer` will create in the DataFrame.
+        /// </summary>
+        /// <param name="value">List of column names which will contain the bucket ID</param>
+        /// <returns>`Bucketizer`</returns>
+        public Bucketizer SetOutputCols(List<string> value)
+        {
+            return WrapAsBucketizer(_jvmObject.Invoke("setOutputCols", value));
+        }
+        
+        /// <summary>
+        /// Executes the `Bucketizer` and transforms the DataFrame to include the new column or
+        /// columns with the bucketed data.
+        /// </summary>
+        /// <param name="source">The DataFrame to add the bucketed data to</param>
+        /// <returns>`DataFrame` containing the original data and the new bucketed columns</returns>
+        public DataFrame Transform(DataFrame source)
+        {
+            return new DataFrame((JvmObjectReference)_jvmObject.Invoke("transform"
+                , source));
+        }
+
+        /// <summary>
+        /// The reference we get back from each call isn't usable unless we wrap it in a new dotnet
+        ///  `Bucketizer`
+        /// </summary>
+        /// <param name="obj">The `JvmObjectReference` to convert into a dotnet `Bucketizer`</param>
+        /// <returns>`Bucketizer`</returns>
+        private static Bucketizer WrapAsBucketizer(object obj)
+        {
+            return new Bucketizer((JvmObjectReference)obj);
+        }
+
+        /// <summary>
+        /// The uid that was used to create the `Bucketizer`. If no `UID` is passed in when creating
+        ///  the `Bucketizer` then a random `UID` is created when the `Bucketizer` is created.
+        /// </summary>
+        /// <returns>string `UID` identifying the `Bucketizer`</returns>
+        public string Uid()
+        {
+            return (string)_jvmObject.Invoke("uid");
+        }
+
+        /// <summary>
+        /// How should the `Bucketizer` handle invalid data, choices are "skip", "error" or "keep"
+        /// </summary>
+        /// <returns>`BucketizerInvalidOptions`</returns>
+        public BucketizerInvalidOptions GetHandleInvalid()
```

Comment:
```suggestion
        public string GetHandleInvalid()
```
Whatever spark returns should be the ground truth, let's not do validations here.